main.go

```diff
@@ -278,9 +278,20 @@ func loop() {
 		}
 	}
 
+	if signatureKey == nil {
+		log.Panicf("signature public key cannot be nil")
+	}
+	if len(*signatureKey) == 0 {
+		log.Panicf("signature public key cannot be empty")
+	}
```

Comment:
I would simplify this check
```suggestion
	if signatureKey == nil || len(*signatureKey) == 0 {
		log.Panicf("signature public key should be set")
	}
```